modules/local/insilicoseq/generate/main.nf

```diff
@@ -0,0 +1,70 @@
+process INSILICOSEQ_GENERATE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/insilicoseq:1.6.0--pyh7cba7a3_0':
+        'biocontainers/insilicoseq:1.6.0--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("*.fastq.gz*"), emit: fastq
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args          = task.ext.args ?: ''
+    def args2         = task.ext.args2 ?: ''
+    def prefix        = task.ext.prefix ?: "${meta.id}"
+    def seed          = task.ext.seed ?: "${meta.seed}"
+    if (fasta) {
+        def is_compressed = fasta.name.endsWith(".gz")
+        def fasta_name    = fasta.name.replace(".gz", "")
+        """
+        seed=\$(echo $seed | sed 's/\\[//g' | sed 's/\\]//g')
```

Comment:
That's actually a leftover from a previous implementation of the pipeline. Before, I had meta and fasta as two separate inputs into the module and I used .first() to turn the fasta channel into a value channel which allowed the module to run multiple times when the samplesheet has multiple rows. The problem is that this method turned the meta elements into lists. So I had this sed step to remove the square brackets. I later figured it would be neater to use .combine() to combine the meta and fasta channels. This means the sed step is not needed anymore. I'll remove it to avoid confusion.